--- a/chrome/browser/ui/views/location_bar/intent_chip_button.cc
+++ b/chrome/browser/ui/views/location_bar/intent_chip_button.cc
@@ -87,14 +87,10 @@ bool IntentChipButton::GetShowChip() con
 }
 
 bool IntentChipButton::GetChipExpanded() const {
-  auto* tab_helper = GetTabHelper();
-  return tab_helper && tab_helper->ShouldShowExpandedChip();
+  return false;
 }
 
 ui::ImageModel IntentChipButton::GetAppIcon() const {
-  if (auto* tab_helper = GetTabHelper()) {
-    return tab_helper->app_icon();
-  }
   return ui::ImageModel();
 }
 
--- a/chrome/browser/ui/views/location_bar/omnibox_chip_button.cc
+++ b/chrome/browser/ui/views/location_bar/omnibox_chip_button.cc
@@ -56,8 +56,7 @@ OmniboxChipButton::OmniboxChipButton(Pre
   SetFocusBehavior(views::View::FocusBehavior::ALWAYS);
   // Equalizing padding on the left, right and between icon and label.
   SetImageLabelSpacing(kChipImagePadding);
-  SetCustomPadding(
-      gfx::Insets::VH(kChipVerticalPadding, kChipHorizontalPadding));
+  SetCustomPadding(GetLayoutInsets(LOCATION_BAR_PAGE_ACTION_ICON_PADDING));
   label()->SetTextStyle(views::style::STYLE_BODY_4_EMPHASIS);
   SetCornerRadius(GetLayoutConstant(LOCATION_BAR_CHILD_CORNER_RADIUS));
   animation_ = std::make_unique<gfx::SlideAnimation>(this);
