--- a/chrome/common/pref_names.h
+++ b/chrome/common/pref_names.h
@@ -1340,6 +1340,26 @@ inline constexpr char kSplitViewDragAndD
 inline constexpr char kSplitViewDragAndDropNudgeUsedCount[] =
     "browser.split_view_drag_and_drop_nudge_used_count";
 
+// A boolean pref set to true if the Back button should be visible on the
+// toolbar.
+inline constexpr char kShowBackButton[] = "helium.browser.show_back_button";
+
+// A boolean pref set to true if the Reload button should be visible on the
+// toolbar.
+inline constexpr char kShowReloadButton[] = "helium.browser.show_reload_button";
+
+// A boolean pref set to true if the Avatar button should be visible on the
+// toolbar.
+inline constexpr char kShowAvatarButton[] = "helium.browser.show_avatar_button";
+
+// A boolean pref set to true if the Extensions button should be visible on the
+// toolbar.
+inline constexpr char kShowExtensionsButton[] = "helium.browser.show_extensions_button";
+
+// A boolean pref set to true if the Menu button should be visible on the
+// toolbar.
+inline constexpr char kShowMenuButton[] = "helium.browser.show_menu_button";
+
 // A boolean pref set to true if Gemini integration be enabled. This is managed
 // by enterprise policy.
 inline constexpr char kGeminiSettings[] = "browser.gemini_settings";
--- a/chrome/browser/ui/browser_ui_prefs.cc
+++ b/chrome/browser/ui/browser_ui_prefs.cc
@@ -130,6 +130,12 @@ void RegisterBrowserUserPrefs(user_prefs
   registry->RegisterBooleanPref(prefs::kPinSplitTabButton, false,
                                 pref_registration_flags);
 
+  registry->RegisterBooleanPref(prefs::kShowBackButton, true);
+  registry->RegisterBooleanPref(prefs::kShowReloadButton, true);
+  registry->RegisterBooleanPref(prefs::kShowAvatarButton, true);
+  registry->RegisterBooleanPref(prefs::kShowExtensionsButton, true);
+  registry->RegisterBooleanPref(prefs::kShowMenuButton, true);
+
   registry->RegisterInt64Pref(prefs::kDefaultBrowserLastDeclined, 0);
   registry->RegisterBooleanPref(prefs::kWebAppCreateOnDesktop, true);
   registry->RegisterBooleanPref(prefs::kWebAppCreateInAppsMenu, true);
--- a/chrome/browser/extensions/api/settings_private/prefs_util.cc
+++ b/chrome/browser/extensions/api/settings_private/prefs_util.cc
@@ -239,6 +239,18 @@ const PrefsUtil::TypedPrefMap& PrefsUtil
   (*s_allowlist)[::prefs::kSplitViewDragAndDropEnabled] =
       settings_api::PrefType::kBoolean;
 
+  // Helium toolbar settings.
+  (*s_allowlist)[::prefs::kShowBackButton] =
+      settings_api::PrefType::kBoolean;
+  (*s_allowlist)[::prefs::kShowReloadButton] =
+      settings_api::PrefType::kBoolean;
+  (*s_allowlist)[::prefs::kShowAvatarButton] =
+      settings_api::PrefType::kBoolean;
+  (*s_allowlist)[::prefs::kShowExtensionsButton] =
+      settings_api::PrefType::kBoolean;
+  (*s_allowlist)[::prefs::kShowMenuButton] =
+      settings_api::PrefType::kBoolean;
+
   // Appearance settings.
   (*s_allowlist)[::prefs::kCurrentThemeID] = settings_api::PrefType::kString;
   (*s_allowlist)[::prefs::kPinnedActions] = settings_api::PrefType::kList;
--- a/chrome/browser/ui/actions/chrome_action_id.h
+++ b/chrome/browser/ui/actions/chrome_action_id.h
@@ -566,6 +566,8 @@
 #define TOOLBAR_PINNABLE_ACTION_IDS \
   E(kActionHome, IDC_HOME) \
   E(kActionForward, IDC_FORWARD) \
+  E(kActionAvatar, IDC_SHOW_AVATAR_MENU) \
+  E(kActionExtensions) \
   E(kActionNewIncognitoWindow, IDC_NEW_INCOGNITO_WINDOW) \
   E(kActionSendSharedTabGroupFeedback, IDC_SEND_SHARED_TAB_GROUP_FEEDBACK) \
   E(kActionShowPasswordManager, IDC_SHOW_PASSWORD_MANAGER) \
--- a/chrome/browser/ui/toolbar/pinned_toolbar/pinned_toolbar_actions_model.cc
+++ b/chrome/browser/ui/toolbar/pinned_toolbar/pinned_toolbar_actions_model.cc
@@ -216,6 +216,11 @@ void PinnedToolbarActionsModel::UpdatePi
 void PinnedToolbarActionsModel::ResetToDefault() {
   pref_service_->ClearPref(prefs::kShowHomeButton);
   pref_service_->ClearPref(prefs::kShowForwardButton);
+  pref_service_->ClearPref(prefs::kShowBackButton);
+  pref_service_->ClearPref(prefs::kShowReloadButton);
+  pref_service_->ClearPref(prefs::kShowAvatarButton);
+  pref_service_->ClearPref(prefs::kShowExtensionsButton);
+  pref_service_->ClearPref(prefs::kShowMenuButton);
   pref_service_->ClearPref(prefs::kPinnedActions);
 }
 
@@ -229,7 +234,24 @@ bool PinnedToolbarActionsModel::IsDefaul
   const bool forward_is_default =
       pref_service_->GetDefaultPrefValue(prefs::kShowForwardButton)
           ->GetBool() == pref_service_->GetBoolean(prefs::kShowForwardButton);
-  return action_are_default && home_is_default && forward_is_default;
+  const bool back_is_default =
+      pref_service_->GetDefaultPrefValue(prefs::kShowBackButton)
+          ->GetBool() == pref_service_->GetBoolean(prefs::kShowBackButton);
+  const bool reload_is_default =
+      pref_service_->GetDefaultPrefValue(prefs::kShowReloadButton)
+          ->GetBool() == pref_service_->GetBoolean(prefs::kShowReloadButton);
+  const bool avatar_is_default =
+      pref_service_->GetDefaultPrefValue(prefs::kShowAvatarButton)
+          ->GetBool() == pref_service_->GetBoolean(prefs::kShowAvatarButton);
+  const bool extensions_is_default =
+      pref_service_->GetDefaultPrefValue(prefs::kShowExtensionsButton)
+          ->GetBool() == pref_service_->GetBoolean(prefs::kShowExtensionsButton);
+  const bool menu_is_default =
+      pref_service_->GetDefaultPrefValue(prefs::kShowMenuButton)
+          ->GetBool() == pref_service_->GetBoolean(prefs::kShowMenuButton);
+  return action_are_default && home_is_default && forward_is_default &&
+         back_is_default && reload_is_default && avatar_is_default &&
+         extensions_is_default && menu_is_default;
 }
 
 void PinnedToolbarActionsModel::MaybeMigrateExistingPinnedStates() {
--- a/chrome/browser/ui/views/toolbar/toolbar_view.cc
+++ b/chrome/browser/ui/views/toolbar/toolbar_view.cc
@@ -460,8 +460,7 @@ void ToolbarView::Init() {
 #else
   // DevTools profiles are OffTheRecord, so hide it there.
   show_avatar_toolbar_button = browser_->profile()->IsIncognitoProfile() ||
-                               browser_->profile()->IsGuestSession() ||
-                               browser_->profile()->IsRegularProfile();
+                               browser_->profile()->IsGuestSession();
 #endif
 
   const std::string sab_value = base::CommandLine::ForCurrentProcess()->
@@ -530,6 +529,48 @@ void ToolbarView::Init() {
 
   home_->SetVisible(show_home_button_.GetValue());
 
+  show_back_button_.Init(
+      prefs::kShowBackButton, prefs,
+      base::BindRepeating(&ToolbarView::OnShowBackButtonChanged,
+                          base::Unretained(this)));
+
+  back_->SetVisible(show_back_button_.GetValue());
+
+  show_reload_button_.Init(
+      prefs::kShowReloadButton, prefs,
+      base::BindRepeating(&ToolbarView::OnShowReloadButtonChanged,
+                          base::Unretained(this)));
+
+  reload_->SetVisible(show_reload_button_.GetValue());
+
+  show_avatar_button_.Init(
+      prefs::kShowAvatarButton, prefs,
+      base::BindRepeating(&ToolbarView::OnShowAvatarButtonChanged,
+                          base::Unretained(this)));
+
+  // Avatar is always shown in incognito and guest profiles,
+  // but in regular profiles it's managed by kShowAvatarButton pref.
+  if (browser_->profile()->IsRegularProfile()) {
+    avatar_->SetVisible(show_avatar_button_.GetValue());
+  }
+
+  if (extensions_container_) {
+    show_extensions_button_.Init(
+        prefs::kShowExtensionsButton, prefs,
+        base::BindRepeating(&ToolbarView::OnShowExtensionsButtonChanged,
+                            base::Unretained(this)));
+    extensions_container_->SetVisible(show_extensions_button_.GetValue());
+  }
+
+  if (app_menu_button_) {
+    show_menu_button_.Init(
+        prefs::kShowMenuButton, prefs,
+        base::BindRepeating(&ToolbarView::OnShowMenuButtonChanged,
+                            base::Unretained(this)));
+
+    app_menu_button_->SetVisible(show_menu_button_.GetValue());
+  }
+
   InitLayout();
 
   for (auto* button : std::array<views::Button*, 5>{back_, forward_, reload_,
@@ -1238,10 +1279,33 @@ void ToolbarView::OnShowForwardButtonCha
   InvalidateLayout();
 }
 
+void ToolbarView::OnShowBackButtonChanged() {
+  back_->SetVisible(show_back_button_.GetValue());
+  InvalidateLayout();
+}
+
+void ToolbarView::OnShowReloadButtonChanged() {
+  reload_->SetVisible(show_reload_button_.GetValue());
+  InvalidateLayout();
+}
+
 void ToolbarView::OnShowHomeButtonChanged() {
   home_->SetVisible(show_home_button_.GetValue());
 }
 
+void ToolbarView::OnShowAvatarButtonChanged() {
+  avatar_->SetVisible(show_avatar_button_.GetValue());
+}
+
+void ToolbarView::OnShowExtensionsButtonChanged() {
+  extensions_container_->SetVisible(show_extensions_button_.GetValue());
+}
+
+void ToolbarView::OnShowMenuButtonChanged() {
+  app_menu_button_->SetVisible(show_menu_button_.GetValue());
+  InvalidateLayout();
+}
+
 void ToolbarView::OnTouchUiChanged() {
   if (display_mode_ == DisplayMode::NORMAL) {
     // Update the internal margins for touch layout.
--- a/chrome/browser/ui/views/toolbar/toolbar_view.h
+++ b/chrome/browser/ui/views/toolbar/toolbar_view.h
@@ -278,6 +278,16 @@ class ToolbarView : public views::Access
 
   void OnShowHomeButtonChanged();
 
+  void OnShowBackButtonChanged();
+
+  void OnShowReloadButtonChanged();
+
+  void OnShowAvatarButtonChanged();
+
+  void OnShowExtensionsButtonChanged();
+
+  void OnShowMenuButtonChanged();
+
   void OnTouchUiChanged();
 
   void UpdateClipPath();
@@ -329,6 +339,16 @@ class ToolbarView : public views::Access
   // Controls whether or not a home button should be shown on the toolbar.
   BooleanPrefMember show_home_button_;
 
+  BooleanPrefMember show_back_button_;
+
+  BooleanPrefMember show_reload_button_;
+
+  BooleanPrefMember show_avatar_button_;
+
+  BooleanPrefMember show_extensions_button_;
+
+  BooleanPrefMember show_menu_button_;
+
   BooleanPrefMember show_chrome_labs_button_;
 
   // The display mode used when laying out the toolbar.
--- a/chrome/browser/ui/webui/side_panel/customize_chrome/customize_toolbar/customize_toolbar.mojom
+++ b/chrome/browser/ui/webui/side_panel/customize_chrome/customize_toolbar/customize_toolbar.mojom
@@ -16,6 +16,11 @@ enum ActionId {
   kShowSearchCompanion,
   kHome,
   kForward,
+  kBack,
+  kReload,
+  kAvatar,
+  kExtensions,
+  kMenu,
   kNewIncognitoWindow,
   kShowPasswordManager,
   kShowPaymentMethods,
--- a/chrome/browser/ui/webui/side_panel/customize_chrome/customize_toolbar/customize_toolbar_handler.cc
+++ b/chrome/browser/ui/webui/side_panel/customize_chrome/customize_toolbar/customize_toolbar_handler.cc
@@ -41,6 +41,16 @@ MojoActionForChromeAction(actions::Actio
       return side_panel::customize_chrome::mojom::ActionId::kHome;
     case kActionForward:
       return side_panel::customize_chrome::mojom::ActionId::kForward;
+    case kActionBack:
+      return side_panel::customize_chrome::mojom::ActionId::kBack;
+    case kActionReload:
+      return side_panel::customize_chrome::mojom::ActionId::kReload;
+    case kActionAvatar:
+      return side_panel::customize_chrome::mojom::ActionId::kAvatar;
+    case kActionExtensions:
+      return side_panel::customize_chrome::mojom::ActionId::kExtensions;
+    case kActionShowAppMenu:
+      return side_panel::customize_chrome::mojom::ActionId::kMenu;
     case kActionNewIncognitoWindow:
       return side_panel::customize_chrome::mojom::ActionId::kNewIncognitoWindow;
     case kActionShowDownloads:
@@ -85,6 +95,16 @@ std::optional<actions::ActionId> ChromeA
       return kActionHome;
     case side_panel::customize_chrome::mojom::ActionId::kForward:
       return kActionForward;
+    case side_panel::customize_chrome::mojom::ActionId::kBack:
+      return kActionBack;
+    case side_panel::customize_chrome::mojom::ActionId::kReload:
+      return kActionReload;
+    case side_panel::customize_chrome::mojom::ActionId::kAvatar:
+      return kActionAvatar;
+    case side_panel::customize_chrome::mojom::ActionId::kExtensions:
+      return kActionExtensions;
+    case side_panel::customize_chrome::mojom::ActionId::kMenu:
+      return kActionShowAppMenu;
     case side_panel::customize_chrome::mojom::ActionId::kNewIncognitoWindow:
       return kActionNewIncognitoWindow;
     case side_panel::customize_chrome::mojom::ActionId::kShowDownloads:
@@ -144,6 +164,26 @@ CustomizeToolbarHandler::CustomizeToolba
       prefs::kPinSplitTabButton,
       base::BindRepeating(&CustomizeToolbarHandler::OnPinSplitTabButtonChanged,
                           base::Unretained(this)));
+  pref_change_registrar_.Add(
+      prefs::kShowBackButton,
+      base::BindRepeating(&CustomizeToolbarHandler::OnShowBackButtonChanged,
+                          base::Unretained(this)));
+  pref_change_registrar_.Add(
+      prefs::kShowReloadButton,
+      base::BindRepeating(&CustomizeToolbarHandler::OnShowReloadButtonChanged,
+                          base::Unretained(this)));
+  pref_change_registrar_.Add(
+      prefs::kShowAvatarButton,
+      base::BindRepeating(&CustomizeToolbarHandler::OnShowAvatarButtonChanged,
+                          base::Unretained(this)));
+  pref_change_registrar_.Add(
+      prefs::kShowExtensionsButton,
+      base::BindRepeating(&CustomizeToolbarHandler::OnShowExtensionsButtonChanged,
+                          base::Unretained(this)));
+  pref_change_registrar_.Add(
+      prefs::kShowMenuButton,
+      base::BindRepeating(&CustomizeToolbarHandler::OnShowMenuButtonChanged,
+                          base::Unretained(this)));
 }
 
 CustomizeToolbarHandler::~CustomizeToolbarHandler() = default;
@@ -186,8 +226,64 @@ void CustomizeToolbarHandler::ListAction
               .Rasterize(&provider),
           scale_factor)));
 
-  actions.push_back(std::move(home_action));
+  auto back_action = side_panel::customize_chrome::mojom::Action::New(
+      MojoActionForChromeAction(kActionBack).value(),
+      base::UTF16ToUTF8(l10n_util::GetStringUTF16(IDS_ACCNAME_BACK)),
+      prefs()->GetBoolean(prefs::kShowBackButton), false,
+      side_panel::customize_chrome::mojom::CategoryId::kNavigation,
+      GURL(webui::EncodePNGAndMakeDataURI(
+          ui::ImageModel::FromVectorIcon(
+              vector_icons::kBackArrowChromeRefreshIcon, icon_color_id)
+              .Rasterize(&provider),
+          scale_factor)));
+
+  auto reload_action = side_panel::customize_chrome::mojom::Action::New(
+      MojoActionForChromeAction(kActionReload).value(),
+      base::UTF16ToUTF8(l10n_util::GetStringUTF16(IDS_ACCNAME_RELOAD)),
+      prefs()->GetBoolean(prefs::kShowReloadButton), false,
+      side_panel::customize_chrome::mojom::CategoryId::kNavigation,
+      GURL(webui::EncodePNGAndMakeDataURI(
+          ui::ImageModel::FromVectorIcon(
+              vector_icons::kReloadChromeRefreshIcon, icon_color_id)
+              .Rasterize(&provider),
+          scale_factor)));
+
+  auto avatar_action = side_panel::customize_chrome::mojom::Action::New(
+      MojoActionForChromeAction(kActionAvatar).value(),
+      base::UTF16ToUTF8(l10n_util::GetStringUTF16(IDS_PROFILES_MENU_NAME)),
+      prefs()->GetBoolean(prefs::kShowAvatarButton), false,
+      side_panel::customize_chrome::mojom::CategoryId::kYourChrome,
+      GURL(webui::EncodePNGAndMakeDataURI(
+          ui::ImageModel::FromVectorIcon(
+              vector_icons::kAccountCircleChromeRefreshIcon, icon_color_id)
+              .Rasterize(&provider),
+          scale_factor)));
+
+  auto extensions_action = side_panel::customize_chrome::mojom::Action::New(
+      MojoActionForChromeAction(kActionExtensions).value(),
+      base::UTF16ToUTF8(l10n_util::GetStringUTF16(IDS_TOOLTIP_EXTENSIONS_BUTTON)),
+      prefs()->GetBoolean(prefs::kShowExtensionsButton), false,
+      side_panel::customize_chrome::mojom::CategoryId::kYourChrome,
+      GURL(webui::EncodePNGAndMakeDataURI(
+          ui::ImageModel::FromVectorIcon(
+              vector_icons::kExtensionChromeRefreshIcon, icon_color_id)
+              .Rasterize(&provider),
+          scale_factor)));
+
+  auto menu_action = side_panel::customize_chrome::mojom::Action::New(
+      MojoActionForChromeAction(kActionShowAppMenu).value(),
+      base::UTF16ToUTF8(l10n_util::GetStringUTF16(IDS_ACCNAME_APP_MENU)),
+      prefs()->GetBoolean(prefs::kShowMenuButton), false,
+      side_panel::customize_chrome::mojom::CategoryId::kYourChrome,
+      GURL(webui::EncodePNGAndMakeDataURI(
+          ui::ImageModel::FromVectorIcon(kBrowserToolsIcon, icon_color_id)
+              .Rasterize(&provider),
+          scale_factor)));
+
+  actions.push_back(std::move(back_action));
   actions.push_back(std::move(forward_action));
+  actions.push_back(std::move(reload_action));
+  actions.push_back(std::move(home_action));
 
   if (base::FeatureList::IsEnabled(features::kSideBySide)) {
     auto split_tab_action = side_panel::customize_chrome::mojom::Action::New(
@@ -203,6 +299,10 @@ void CustomizeToolbarHandler::ListAction
     actions.push_back(std::move(split_tab_action));
   }
 
+  actions.push_back(std::move(extensions_action));
+  actions.push_back(std::move(avatar_action));
+  actions.push_back(std::move(menu_action));
+
   const auto add_action =
       [&actions, this, &provider, scale_factor, bwi](
           actions::ActionId id,
@@ -258,17 +358,16 @@ void CustomizeToolbarHandler::ListAction
         actions.push_back(std::move(mojo_action));
       };
 
-  add_action(kActionNewIncognitoWindow,
-             side_panel::customize_chrome::mojom::CategoryId::kNavigation);
-
   add_action(kActionSidePanelShowBookmarks,
-             side_panel::customize_chrome::mojom::CategoryId::kYourChrome);
+             side_panel::customize_chrome::mojom::CategoryId::kTools);
   add_action(kActionSidePanelShowHistoryCluster,
-             side_panel::customize_chrome::mojom::CategoryId::kYourChrome);
+             side_panel::customize_chrome::mojom::CategoryId::kTools);
   add_action(kActionShowDownloads,
-             side_panel::customize_chrome::mojom::CategoryId::kYourChrome);
+             side_panel::customize_chrome::mojom::CategoryId::kTools);
   add_action(kActionClearBrowsingData,
-             side_panel::customize_chrome::mojom::CategoryId::kYourChrome);
+             side_panel::customize_chrome::mojom::CategoryId::kTools);
+  add_action(kActionNewIncognitoWindow,
+             side_panel::customize_chrome::mojom::CategoryId::kTools);
 
   if (features::HasTabSearchToolbarButton()) {
     add_action(kActionTabSearch,
@@ -334,6 +433,21 @@ void CustomizeToolbarHandler::PinAction(
     case kActionSplitTab:
       prefs()->SetBoolean(prefs::kPinSplitTabButton, pin);
       break;
+    case kActionBack:
+      prefs()->SetBoolean(prefs::kShowBackButton, pin);
+      break;
+    case kActionReload:
+      prefs()->SetBoolean(prefs::kShowReloadButton, pin);
+      break;
+    case kActionAvatar:
+      prefs()->SetBoolean(prefs::kShowAvatarButton, pin);
+      break;
+    case kActionExtensions:
+      prefs()->SetBoolean(prefs::kShowExtensionsButton, pin);
+      break;
+    case kActionShowAppMenu:
+      prefs()->SetBoolean(prefs::kShowMenuButton, pin);
+      break;
     default:
       model_->UpdatePinnedState(chrome_action.value(), pin);
       const std::optional<std::string> metrics_name =
@@ -387,6 +501,31 @@ void CustomizeToolbarHandler::OnPinSplit
                         prefs()->GetBoolean(prefs::kPinSplitTabButton));
 }
 
+void CustomizeToolbarHandler::OnShowBackButtonChanged() {
+  OnActionPinnedChanged(kActionBack,
+                        prefs()->GetBoolean(prefs::kShowBackButton));
+}
+
+void CustomizeToolbarHandler::OnShowReloadButtonChanged() {
+  OnActionPinnedChanged(kActionReload,
+                        prefs()->GetBoolean(prefs::kShowReloadButton));
+}
+
+void CustomizeToolbarHandler::OnShowAvatarButtonChanged() {
+  OnActionPinnedChanged(kActionAvatar,
+                        prefs()->GetBoolean(prefs::kShowAvatarButton));
+}
+
+void CustomizeToolbarHandler::OnShowExtensionsButtonChanged() {
+  OnActionPinnedChanged(kActionExtensions,
+                        prefs()->GetBoolean(prefs::kShowExtensionsButton));
+}
+
+void CustomizeToolbarHandler::OnShowMenuButtonChanged() {
+  OnActionPinnedChanged(kActionShowAppMenu,
+                        prefs()->GetBoolean(prefs::kShowMenuButton));
+}
+
 void CustomizeToolbarHandler::OnActionItemChanged() {
   client_->NotifyActionsUpdated();
 }
--- a/chrome/browser/ui/webui/side_panel/customize_chrome/customize_toolbar/customize_toolbar_handler.h
+++ b/chrome/browser/ui/webui/side_panel/customize_chrome/customize_toolbar/customize_toolbar_handler.h
@@ -54,6 +54,11 @@ class CustomizeToolbarHandler
   void OnShowHomeButtonChanged();
   void OnShowForwardButtonChanged();
   void OnPinSplitTabButtonChanged();
+  void OnShowBackButtonChanged();
+  void OnShowReloadButtonChanged();
+  void OnShowAvatarButtonChanged();
+  void OnShowExtensionsButtonChanged();
+  void OnShowMenuButtonChanged();
   void OnActionItemChanged();
 
   PrefService* prefs() const;
--- a/chrome/browser/ui/views/extensions/extensions_toolbar_container.cc
+++ b/chrome/browser/ui/views/extensions/extensions_toolbar_container.cc
@@ -554,7 +554,9 @@ void ExtensionsToolbarContainer::AnchorA
 
   // Fix the position of widgets. Without this fix, extension-installed-bubble
   // and extension-uninstall-dialog may be out of the window border on Linux.
-  if (base::CommandLine::ForCurrentProcess()->HasSwitch("hide-extensions-menu"))
+  if (!browser_->profile()->GetPrefs()->GetBoolean(
+      prefs::kShowExtensionsButton) ||
+      base::CommandLine::ForCurrentProcess()->HasSwitch("hide-extensions-menu"))
   {
     views::View* anchor_view = BrowserView::GetBrowserViewForBrowser(browser_)
       ->toolbar_button_provider()->GetAppMenuButton();
@@ -966,7 +968,9 @@ void ExtensionsToolbarContainer::UpdateC
 }
 
 bool ExtensionsToolbarContainer::ShouldContainerBeVisible() const {
-  if (base::CommandLine::ForCurrentProcess()->HasSwitch("hide-extensions-menu"))
+  if (!browser_->profile()->GetPrefs()->GetBoolean(
+      prefs::kShowExtensionsButton) ||
+      base::CommandLine::ForCurrentProcess()->HasSwitch("hide-extensions-menu"))
     return false;
 
   // The container (and extensions-menu button) should not be visible if we have
--- a/chrome/app/generated_resources.grd
+++ b/chrome/app/generated_resources.grd
@@ -10469,6 +10469,9 @@ Keep your key file in a safe place. You
       <message name="IDS_ACCNAME_FIND" desc="The accessible name for the find button.">
         Find
       </message>
+      <message name="IDS_ACCNAME_APP_MENU" desc="The accessible name for the app menu button.">
+        Main Menu
+      </message>
       <message name="IDS_ACCNAME_BOOKMARKS" desc="The accessible name for the Bookmarks toolbar.">
         Bookmarks
       </message>
