--- a/chrome/BUILD.gn
+++ b/chrome/BUILD.gn
@@ -1809,7 +1809,7 @@ if (is_linux || is_chromeos) {
     ]
     if (is_linux) {
       sources += [
-        "//chrome/app/theme/$branding_path_component/linux/product_logo_48.png",
+        "//chrome/app/theme/$branding_path_component/linux/product_logo_256.png",
       ]
     } else {
       sources +=
--- a/chrome/common/channel_info_posix.cc
+++ b/chrome/common/channel_info_posix.cc
@@ -81,18 +81,13 @@ std::string GetChannelName(WithExtendedS
 }
 
 std::string GetChannelSuffixForDataDir() {
-  switch (GetChannel()) {
-    case version_info::Channel::BETA:
-      return "-beta";
-    case version_info::Channel::DEV:
-      return "-unstable";
-    case version_info::Channel::CANARY:
-      return "-canary";
-    default:
+#if !defined(OFFICIAL_BUILD)
+  return ".dev";
+#else
       // Stable, extended stable, and unknown (e.g. in unbranded builds) don't
       // get a suffix.
       return std::string();
-  }
+#endif
 }
 
 #if BUILDFLAG(IS_LINUX)
@@ -148,11 +143,11 @@ std::string GetDesktopName(base::Environ
   // Allow $CHROME_DESKTOP to override the built-in value, so that development
   // versions can set themselves as the default without interfering with
   // non-official, packaged versions using the built-in value.
-  std::optional<std::string> name = env->GetVar("CHROME_DESKTOP");
+  std::optional<std::string> name = env->GetVar("PLUTONIUM_DESKTOP");
   if (name.has_value() && !name.value().empty()) {
     return name.value();
   }
-  return "chromium-browser.desktop";
+  return "plutonium.desktop";
 #endif
 }
 #endif  // BUILDFLAG(IS_LINUX)
--- a/chrome/common/chrome_paths_linux.cc
+++ b/chrome/common/chrome_paths_linux.cc
@@ -81,7 +81,7 @@ bool GetDefaultUserDataDirectory(base::F
   std::unique_ptr<base::Environment> env(base::Environment::Create());
   base::FilePath config_dir;
   std::optional<std::string> chrome_config_home_str =
-      env->GetVar("CHROME_CONFIG_HOME");
+      env->GetVar("PLUTONIUM_CONFIG_HOME");
   if (chrome_config_home_str.has_value() &&
       base::IsStringUTF8(chrome_config_home_str.value())) {
     config_dir = base::FilePath::FromUTF8Unsafe(chrome_config_home_str.value());
@@ -90,13 +90,7 @@ bool GetDefaultUserDataDirectory(base::F
         GetXDGDirectory(env.get(), kXdgConfigHomeEnvVar, kDotConfigDir);
   }
 
-#if BUILDFLAG(GOOGLE_CHROME_FOR_TESTING_BRANDING)
-  std::string data_dir_basename = "google-chrome-for-testing";
-#elif BUILDFLAG(GOOGLE_CHROME_BRANDING)
-  std::string data_dir_basename = "google-chrome";
-#else
-  std::string data_dir_basename = "chromium";
-#endif
+  std::string data_dir_basename = "net.imput.plutonium";
   *result = config_dir.Append(data_dir_basename + GetChannelSuffixForDataDir());
   return true;
 }
--- a/chrome/installer/linux/common/chromium-browser.info
+++ b/chrome/installer/linux/common/chromium-browser.info
@@ -18,7 +18,7 @@ ENROLLMENTDIR=/etc/chromium/policies/enrollment
 
 # Display string for desktop menu/icon.
-MENUNAME="Chromium Web Browser"
+MENUNAME="Plutonium"
 
 # Brief package description.
-SHORTDESC="The web browser from the Chromium projects"
+SHORTDESC="Plutonium web browser"
--- a/media/audio/pulse/pulse_util.cc
+++ b/media/audio/pulse/pulse_util.cc
@@ -39,8 +39,8 @@ namespace {
 constexpr char kBrowserDisplayName[] = "google-chrome";
 #define PRODUCT_STRING "Google Chrome"
 #else
-constexpr char kBrowserDisplayName[] = "chromium-browser";
-#define PRODUCT_STRING "Chromium"
+constexpr char kBrowserDisplayName[] = "plutonium";
+#define PRODUCT_STRING "Plutonium"
 #endif
 
 #if defined(DLOPEN_PULSEAUDIO)
