--- a/components/remote_cocoa/app_shim/native_widget_mac_nswindow.mm
+++ b/components/remote_cocoa/app_shim/native_widget_mac_nswindow.mm
@@ -352,8 +352,12 @@ struct NSEdgeAndCornerThicknesses {
 }
 
 - (NSRect)constrainFrameRect:(NSRect)frameRect toScreen:(NSScreen*)screen {
-  // Headless windows should not be constrained within the physical screen.
-  if (_isHeadless) {
+  if (_isHeadless || self.parentWindow) {
+    // AppKit's default implementation moves child windows down to avoid
+    // the menu bar. We don't want that behavior, because widgets like the
+    // Omnibox may have a big shadow that could cause invisible menu bar
+    // collision in fullscreen/maximized state. We override it here to
+    // return the original frameRect before the adjustment.
     return frameRect;
   }
 
--- a/components/remote_cocoa/app_shim/native_widget_mac_overlay_nswindow.mm
+++ b/components/remote_cocoa/app_shim/native_widget_mac_overlay_nswindow.mm
@@ -6,8 +6,6 @@
 
 #include <AppKit/AppKit.h>
 
-#include "base/mac/mac_util.h"
-
 // Overlay windows are used to render top chrome components while in immersive
 // fullscreen. Those rendered views are hosted in a separate AppKit controlled
 // window. The overlay windows are a transparent overlay on top of the AppKit
@@ -28,19 +26,6 @@ constexpr unsigned long kAllowedCollecti
                                kAllowedCollectionBehaviorMask];
 }
 
-- (NSRect)constrainFrameRect:(NSRect)frameRect toScreen:(NSScreen*)screen {
-  if (base::mac::MacOSVersion() >= 26'00'00) {
-    // Since macOS 26.0, when the main window enters full screen, the
-    // overlay child window’s position is automatically adjusted by the system
-    // to just below the menu bar location (even though the menu bar is hidden
-    // at that time). During this process, [NSWindow constrainFrameRect] is
-    // called and returns the system-adjusted position. Override this
-    // function to return the original value frameRect before the adjustment.
-    return frameRect;
-  }
-  return [super constrainFrameRect:frameRect toScreen:screen];
-}
-
 // Paint the window a mostly opaque color.
 - (void)debugWithColor:(NSColor*)color {
   self.backgroundColor = color;
