--- a/extensions/browser/extension_registrar.cc
+++ b/extensions/browser/extension_registrar.cc
@@ -532,9 +532,12 @@ void ExtensionRegistrar::AddComponentExt
     if (old_version.IsValid()) {
       UnregisterServiceWorkerWithRootScope(extension);
     }
+
+    base::flat_set<int> disable_reasons = GetDisableReasonsOnInstalled(extension);
+
     // TODO(crbug.com/40508457): If needed, add support for Declarative Net
     // Request to component extensions and pass the ruleset install prefs here.
-    AddNewOrUpdatedExtension(extension, {}, kInstallFlagNone,
+    AddNewOrUpdatedExtension(extension, std::move(disable_reasons), kInstallFlagNone,
                              syncer::StringOrdinal(), std::string(),
                              /*ruleset_install_prefs=*/{});
     return;
